### PR TITLE
Components: Deprecate `reduceMotion` util

### PR DESCRIPTION
**Pull request by mirka**
## What?

Deprecates the internal `reduceMotion` utility function in the components package.

## Why?

First discussed in https://github.com/WordPress/gutenberg/pull/60560#discussion_r1560895894.

- Adds unnecessary abstraction.
- Does not allow for more nuanced styling, such as reducing motion instead of completely removing it.
- In cases like where this is already used, it would be simpler to wrap the motion-related styles in a `@media not ( prefers-reduced-motion )` block.

It isn't particularly important to remove the existing usages right now — they can be rewritten as needed. We'll be able to remove this util from the codebase once they're all rewritten though.

## Testing Instructions

In your IDE, existing usages of this util should show that it is deprecated.

In VS Code, usages of the util are struck out to flag as deprecated

### CODE REVIEW

**Review by Mamaduka — APPROVED** [review]
Makes sense 👍

Thank you, @mirka!

File: packages/components/src/utils/reduce-motion.js [hunk 1]
```diff
@@ -3,6 +3,10 @@
  *
  * @param {'transition' | 'animation' | string} [prop='transition'] CSS Property name
  * @return {string} Generated CSS code for the reduced style
+ *
+ * @deprecated Write your own media query instead,
```
Comment (tyxla) [inline packages/components/src/utils/reduce-motion.js:7]: Not in this PR, but should we also refactor the existing usages? The best way to ensure that there will be new usage of this deprecated utility is to keep using it in the Gutenberg codebase.

**Review by tyxla — APPROVED** [review]
Good call 👍

File: packages/components/src/utils/reduce-motion.js [hunk 2]
```diff
@@ -3,6 +3,10 @@
...
+ * e.g. `@media not ( prefers-reduced-motion ) { ...some animation... }` or
```
Comment (DaniGuardiola) [inline packages/components/src/utils/reduce-motion.js:8]: Maybe this is just a typo or maybe it's another valid way to implement this, but:

```diff
-@media not ( prefers-reduced-motion ) {
+@media not ( prefers-reduced-motion: reduce ) {
```

Are we sure those will behave the same way with no weird edge cases? Is there any practical difference between these two that could be relevant?

**Review by DaniGuardiola — APPROVED** [review]
Very nice! Only a small question.

File: packages/components/src/utils/reduce-motion.js (hunk 1 above)
Comment (DaniGuardiola) [reply packages/components/src/utils/reduce-motion.js:7]: +1 ^

File: packages/components/src/utils/reduce-motion.js (hunk 2 above)
Comment (mirka) [reply packages/components/src/utils/reduce-motion.js:8]: The [current docs](https://drafts.csswg.org/mediaqueries-5/#mq-boolean-context) do say they should be equivalent. They would differ if they ever added another possible value for [`prefers-reduced-motion`](https://drafts.csswg.org/mediaqueries-5/#prefers-reduced-motion), for example `reduce-somewhat` or `absolutely-no-motion-im-serious`, but in that case I think it'd be better for those values to be included in our simple booleans by default.

File: packages/components/src/utils/reduce-motion.js (hunk 1 above)
Comment (mirka) [reply packages/components/src/utils/reduce-motion.js:7]: It's just a matter of prioritization. Ideally they should be rewritten, but this one is pretty benign compared to the million other things we need to rewrite 😭 Though, it's probably a good first issue for new contributors so I'll post an issue.

File: packages/components/src/utils/reduce-motion.js (hunk 2 above)
Comment (DaniGuardiola) [reply packages/components/src/utils/reduce-motion.js:8]: The linked spec section is a very interesting read, and makes a lot of sense 👍

File: packages/components/src/utils/reduce-motion.js (hunk 1 above)
Comment (tyxla) [reply packages/components/src/utils/reduce-motion.js:7]: No urgency here, an issue sounds good 👍

**Pull request merged**